app/data/tasks/deduplicate_records.py

```diff
@@ -0,0 +1,192 @@
+from django.db.models import Q
+
+from celery import shared_task
+from celery.app.task import Task
+from celery.utils.log import get_task_logger
+
+from data.models import (DedupeJob, RecordDuplicate)
+from ashlar.models import Record
+
+import datetime
+from pytz import timezone
+
+logger = get_task_logger(__name__)
+
+
+class DedupeTask(Task):
+    """Base class for the deduplication task to handle updating status in database"""
+
+    def on_success(self, retval, task_id, args, kwargs):
+        job = DedupeJob.objects.get(celery_task=task_id)
+        job.status = DedupeJob.Status.SUCCESS
+        job.save()
+
+    def on_failure(self, exc, task_id, args, kwargs, einfo):
+        """ On failure, delete duplicates found in the unsuccessful run"""
+        job = DedupeJob.objects.get(celery_task=task_id)
+        job.status = DedupeJob.Status.ERROR
+        job.save()
+        RecordDuplicate.objects.filter(job=job).delete()
+
+
+def calculate_similarity_score(record1, record2, time_allowance, distance_allowance):
+    """
+    Set a score representing the likelyhood the two records are duplicated, based
+    on the criteria.  score is a float between 0 and 1
+
+    normalized dimensions are considered equal (1 day equally as important as 100m)
+
+    If the records are more than time_allowance or distance_allowance apart, this will
+    return a negative value
+    """
+    tdelta = abs(record1.occurred_from - record2.occurred_from)
+    tscore = 1.0 - (tdelta.total_seconds() / time_allowance.total_seconds())
+    ddelta = record1.geom.distance(record2.geom)
+    dscore = 1.0 - (ddelta / distance_allowance)
+    score = (tscore + dscore) / 2
+    return score
+
+
+def find_duplicates_for_record(uuid, time_allowance, distance_allowance):
+    """ Given a record, find all possible duplicates for it
+    """
+    record = Record.objects.get(uuid=uuid)
+    return Record.objects.filter(
+        schema__record_type=record.schema.record_type,
+        geom__dwithin=(record.geom, distance_allowance),
+        occurred_from__range=(
+            record.occurred_from - time_allowance, record.occurred_from + time_allowance
+        )
+    ).exclude(
+        Q(uuid=record.uuid) | Q(
+            uuid__in=RecordDuplicate.objects.filter(
+                Q(record__uuid=record.uuid) | Q(
+                    duplicate_record__uuid=record.uuid
+                )
+            )
+        )
+    )
+
+
+def get_time_extent(job):
+    """ Get the time range to dedupe
+    Calculate from X hr before the last dedupe run to the current task run's
+    start time, where X is the heuristic's time allowance
+
+    Returns:
+        dictionary containing the start and end times for the next dedupe task
+        {starttime: timestamp, endtime: timestamp}
+    """
+    last_job = None
+    if DedupeJob.objects.all().count() > 0:
+        try:
+            last_job = DedupeJob.objects.exclude(uuid=job.uuid).latest()
+        except:
```

Comment:
If there's an exception in particular that you have in mind here, it's better to catch that specifically; a blanket
```python
except:
    pass
```
can be really tough to debug.